pkg/controller/deployment/deployment_controller.go

```diff
@@ -581,6 +581,20 @@ func (dc *DeploymentController) syncDeployment(key string) error {
 		return nil
 	}
 
+	// This is the point at which we used to add/remove the overlap annotation.
```

Comment:
Filed #43322. Added comment in another commit, which I'll send as a separate PR unless this one requires fixup before merging.